docs/cookbook/named_entity_recognition.md

```diff
@@ -0,0 +1,422 @@
+# Named Entity Recognition
+
+In this guide, we demonstrate techniques to perform Named Entity Recognition (NER) using Large Language Models (LLMs). We will use Groq's llama-3.1-8b-instant model, but you can adapt this approach to other models with similar capabilities.
+
+??? tip "Mirascope Concepts Used"
+
+    - [Calls](../learn/calls.md)
+    - [Response Model](../learn/response_models.md)
+    - [Prompt Templates](../learn/prompt_templates.md)
+
+!!! note "Background"
+
+    Named Entity Recognition is a subtask of information extraction that seeks to locate and classify named entities in text into predefined categories such as person names, organizations, locations, etc. LLMs have revolutionized NER by enabling more context-aware and hierarchical entity recognition, going beyond traditional rule-based or statistical methods.
+
+## Setup and Imports
```

Comment:
nit: take a look at the structure of the other cookbooks here. For setup, we should show installing the necessary packages and ensuring api keys are set.

we try to keep imports in the code cells where they are used to it's easier to grok what's going on

Reply:
Yes, I had written this in little hurry. Will fix it soon.

Reply:
Added structure and instructions